.github/workflows/build.yaml

```diff
@@ -0,0 +1,37 @@
+name: build
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: install pre-commit
+        run: pip install pre-commit
+      - name: pre-commit
+        run: pre-commit run --all-files --show-diff-on-failure
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get kubebuilder
+        uses: RyanSiu1995/kubebuilder-action@v1.3.1
+        with:
+          version: 4.1.1
```

Comment:
Managed to get integration tests working on the CI using this nice GH actions plugin for `kubebuilder`

Will figure out and document how to set it up locally as part of https://github.com/pelotech/nidhogg/issues/23

Reply:
nice find :)